src/EVM.hs

```diff
@@ -672,13 +672,8 @@ exec1 conf = do
             _ -> underrun
             where
             mcopy sz srcOff dstOff = do
-                  m <- gets (.state.memory)
-                  case m of
-                    ConcreteMemory mem -> do
-                      buf <- freezeMemory mem
-                      copyBytesToMemory buf sz srcOff dstOff
-                    SymbolicMemory mem -> do
-                      assign (#state % #memory) (SymbolicMemory $ copySlice srcOff dstOff sz mem mem)
```

Comment:
I think this previous code results in a bit simpler representation for the symbolic case, so we might want to keep it and only modify the concrete case